configuration.md

```diff
@@ -0,0 +1,104 @@
+# Proposal for configuring supported
+
+Teams may want to customize the parameters of a support policy. For instance, a team may want to ignore certain dependencies, i.e. not enforce upgrades for certain dependencies. They may also want different upgrade cadences for specific dependencies. Lastly, in many cases, simply turning on a support policy and expecting consumers to immediately adhere to the policy is untenable (for instance, a major upgrade with a migration may be emminently required shortly after turning on a support policy). There needs to be a way to provide a grace period for onboarding to the support policy, while still providing a warning for when dependencies must be updated.
+
+We will not require a configuration file, and will default to 4, 2, and 1 quarter upgrade periods for major, minor, and patch versions respectively.
+
+```JSON
+{
+  "primary": {
+    "upgradeBudget": {
+      "major": 4,
+      "minor": 4,
+      "patch": 2
+    }
+  }
+}
+```
+
+```TypeScript
+type Quarter = number;
+type Weeks = number;
+type DateString = string // month/day/year for instance "12/25/2021"
+
+interface UpgradeBudget {
+  major: Quarters;
+  minor: Quarters;
+  patch: Quarters;
+
+  // Any releases that happen within this number of weeks of the end of a quarter
+  // are considered to have been released the next quarter
+  tailGracePeriod?: Weeks;
+}
+
+interface Policy {
+  // Fallback to default 4, 2, 1 budget with no tailGracePeriod.
+  upgradeBudget?: UpgradeBudget;
+  
+  // Used for easing users into a new policy: the support policy will act as
+  // if all versions of dependencies were released on this date if the actual
+  // release date is before the date. Users will be warned of the effective
+  // deprecation date whenever the support policy tool is run.
+  effectiveReleaseDate?: DateString;
+}
+
+interface PrimaryPolicy extends Policy {
+  // If we ignore a dependency, that means that the given package will not affect
+  // whether or not a consumer is within the support policy
+  ignoredDependencies?: string[];
+}
+
+interface CustomPolicy extends Policy {
+  // A whitelist of packages that are looked at for this policy
+  dependencies: string[];
```

Comment:
Are we targeting to override something like if pre-release present and we should generally ignore by default, but if we want to include, this could be a place where we can set it.

Reply:
no, custom policies do not allow changing behavior for pre-releases specifically.